src/backend/access/zedstore/zedstore_btree.c

```diff
@@ -146,17 +136,66 @@ zsbt_end_scan(ZSBtreeScan *scan)
 	scan->active = false;
 }
 
+static void
+zsbt_fill_missing_attribute_value(ZSBtreeScan *scan, Datum *datum, bool *isnull)
+{
+	int attno = scan->attno - 1;
+	AttrMissing *attrmiss = NULL;
+	TupleDesc tupleDesc = scan->rel->rd_att;
+	Form_pg_attribute attr = TupleDescAttr(tupleDesc, attno);
+
+	*isnull = true;
+	if (tupleDesc->constr &&
+		tupleDesc->constr->missing)
+	{
+		/*
+		 * If there are missing values we want to put them into the
+		 * tuple.
+		 */
+		attrmiss = tupleDesc->constr->missing;
+
+		if (attrmiss[attno].am_present)
+		{
+			*isnull = false;
+			if (attr->attbyval)
+				*datum = fetch_att(&attrmiss[attno].am_value, attr->attbyval, attr->attlen);
+			else
+				*datum = zs_datumCopy(attrmiss[attno].am_value, attr->attbyval, attr->attlen);
+		}
+	}
+}
+
 /*
  * Return true if there was another tuple. The datum is returned in *datum,
  * and its TID in *tid. For a pass-by-ref datum, it's a palloc'd copy.
  */
 bool
-zsbt_scan_next(ZSBtreeScan *scan, Datum *datum, bool *isnull, zstid *tid)
+zsbt_scan_next(ZSBtreeScan *scan, Datum *datum, bool *isnull, zstid *tid, bool *isvaluemissing)
```

Comment:
With so many return parameters, would it make sense to return a struct here, or is this following a pattern?